File: 1704-determine-if-string-halves-are-alike/1704-determine-if-string-halves-are-alike.cpp
class Solution {
public:
    bool halvesAreAlike(string s) {
        int cnt=0;
        for(int i=0; i<s.size(); i++){
            if(s[i]=='a' || s[i]=='A' || s[i]=='e' || s[i]=='E' || s[i]=='i' || s[i]=='I' || s[i]=='o' || s[i]=='O' || s[i]=='u' || s[i]=='U'){
                if(i<s.length()/2){
                    cnt++;
                }else cnt--;
            }
        }
        return cnt==0;
    }
};

File: 1704-determine-if-string-halves-are-alike/1704-determine-if-string-halves-are-alike.java
class Solution {
    public boolean isVowel(char ch){
        return ch=='a' || ch=='e' || ch=='i' || ch=='o' || ch=='u' || ch=='A'|| ch=='E' || ch=='I' || ch=='O' || ch=='U';
    }
    public boolean halvesAreAlike(String s) {
        int cnt=0;
        for(int i=0; i<s.length(); i++){
            if(isVowel(s.charAt(i))){
                if(i< s.length()/2) cnt++;
                else cnt--;
            }
        }
        return cnt==0;
    }
}